### PR TITLE
[tests-only][full-ci] removing the setResponse() and $this->response in given/then step in Sharing file

### PR DESCRIPTION
**Pull request by ghost**
## Description
We have used setResponse() and $this->response in the Given/Then steps and some helper functions (maybe to reuse existing available methods). But storing responses from Given/Then steps and helper functions is not a good idea because it can lead to a false positive assertion in the Then steps.
So, check the use of setResponse() and $this->response in
- Given steps
- Then steps (Then steps can use $this->response but must prevent saving to it)
- Helper functions

So this pr make the above changes in `Sharing.php`
## Related Issue
part of: https://github.com/owncloud/ocis/issues/7082

## Motivation and Context
- To  remove setResponse() and $this->response in the Given/Then steps and some helper functions
- To avoid false positive assertions

## How Has This Been Tested?

- test environment:
- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [X] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 1]
```diff
@@ -638,7 +638,7 @@ public function theUserUpdatesTheLastShareWith(?TableNode $body):void {
 	 * @throws Exception
 	 */
 	public function theUserHasUpdatedTheLastShareWith(?TableNode $body):void {
-		$this->updateLastShareByCurrentUser($body);
+		$response = $this->updateLastShareByCurrentUser($body);
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/bootstrap/Sharing.php:641]: why saving response in $response and not using it?

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 2]
```diff
@@ -756,7 +756,7 @@ public function userUpdatesTheLastPublicLinkShareWith(string $user, ?TableNode $
 	 * @throws Exception
 	 */
 	public function userHasUpdatedTheLastShareWith(string $user, ?TableNode $body):void {
-		$this->updateLastShareWithSettings($user, $body);
+		$response = $this->updateLastShareWithSettings($user, $body);
 		$this->theHTTPStatusCodeShouldBeSuccess();
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Sharing.php:760]:
```suggestion
		$this->theHTTPStatusCodeShouldBeBetween(200, 299, $response);
```

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 3]
```diff
@@ -2564,10 +2563,10 @@ public function checkFieldsNotInResponse(?TableNode $body):void {
 	 * @param string $user
 	 * @param string $fileName
 	 *
-	 * @return void
+	 * @return ResponseInterface
 	 * @throws Exception
 	 */
-	public function removeAllSharesFromResource(string $user, string $fileName):void {
+	public function removeAllSharesFromResource(string $user, string $fileName):ResponseInterface {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Sharing.php:2569]: Could you remove this function and the step defs that uses it

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 4]
```diff
@@ -626,7 +626,7 @@ public function updateLastShareByCurrentUser(?TableNode $body):void {
 	 * @throws Exception
 	 */
 	public function theUserUpdatesTheLastShareWith(?TableNode $body):void {
-		$this->updateLastShareByCurrentUser($body);
+		$this->response = $this->updateLastShareByCurrentUser($body);
```
Comment (S-Panta) [inline tests/acceptance/features/bootstrap/Sharing.php:629]:
```suggestion
		$this->setResponse($this->updateLastShareByCurrentUser($body));
```
IMO, this setResponse was created just for this sort of steps

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 5]
```diff
@@ -728,7 +728,7 @@ public function updateLastShareWithSettings(
 	 * @throws Exception
 	 */
 	public function userUpdatesTheLastShareWith(string $user, ?TableNode $body):void {
-		$this->updateLastShareWithSettings($user, $body);
+		$this->response = $this->updateLastShareWithSettings($user, $body);
```
Comment (S-Panta) [inline tests/acceptance/features/bootstrap/Sharing.php:731]:
```suggestion
		$this->setResponse($this->updateLastShareWithSettings($user, $body));
```

**Review by saw-jan — APPROVED** [review]

**Pull request merged**